rx-java2/src/main/java/io/vertx/reactivex/core/RxHelper.java

```diff
@@ -56,9 +56,15 @@ public static Scheduler blockingScheduler(Vertx vertx, boolean ordered) {
    * @return the response observable
    */
   public static Single<String> deployVerticle(Vertx vertx, Verticle verticle, DeploymentOptions options) {
-    return Single.unsafeCreate(new io.vertx.reactivex.core.impl.AsyncResultSingle<String>(handler -> {
-      vertx.getDelegate().deployVerticle(verticle, options, handler);
-    }));
+    return Single.create(emitter -> {
```

Comment:
what does it change to use this instead of the previous construct ?

Reply:
It's the default choice for bridging callback/reactive code. RxJava adds some safety checks with respect to threading and also manages cancellation. You only have to deal with `onError` and `onSuccess`.

Reply:
why then not use instead:

```
  public static Single<String> deployVerticle(Vertx vertx, Verticle verticle, DeploymentOptions options) {
    return new io.vertx.reactivex.core.impl.AsyncResultSingle<String>(handler -> {
      vertx.getDelegate().deployVerticle(verticle, options, handler);
    });
  }
```

Reply:
I had not thought about it. I would do just as well, I think (its code more or less does what `io.reactivex.internal.operators.single.SingleCreate` does). I can change the impl for consistency if you want.

Reply:
yes I would rather use the same scheme, alors it's bit less code

Reply:
Done